lib/srv/db/ca.go

```diff
@@ -205,54 +317,98 @@ func (d *realDownloader) Download(ctx context.Context, database types.Database,
 		} else if strings.HasSuffix(azureCAURLDigiCert, hint) {
 			return d.downloadFromURL(azureCAURLDigiCert)
 		}
-		return nil, trace.BadParameter("unknown Azure CA %q", hint)
+		return nil, nil, trace.BadParameter("unknown Azure CA %q", hint)
 	case types.DatabaseTypeAWSKeyspaces:
 		return d.downloadFromURL(amazonKeyspacesCAURL)
 	}
-	return nil, trace.BadParameter("%v doesn't support automatic CA download", database)
+	return nil, nil, trace.BadParameter("%v doesn't support automatic CA download", database)
+}
+
+// GetVersion returns the CA version for the provided database.
+func (d *realDownloader) GetVersion(ctx context.Context, database types.Database, hint string) ([]byte, error) {
+	switch database.GetType() {
+	case types.DatabaseTypeRDS:
+		return d.getVersionFromURL(database, rdsCAURLForDatabase(database))
+	case types.DatabaseTypeRedshift:
+		return d.getVersionFromURL(database, redshiftCAURLForDatabase(database))
+	case types.DatabaseTypeElastiCache,
+		types.DatabaseTypeMemoryDB:
+		return d.getVersionFromURL(database, amazonRootCA1URL)
+	case types.DatabaseTypeAzure:
+		if strings.HasSuffix(azureCAURLBaltimore, hint) {
+			return d.getVersionFromURL(database, azureCAURLBaltimore)
+		} else if strings.HasSuffix(azureCAURLDigiCert, hint) {
+			return d.getVersionFromURL(database, azureCAURLDigiCert)
+		}
+		return nil, trace.BadParameter("unknown Azure CA %q", hint)
+	case types.DatabaseTypeAWSKeyspaces:
+		return d.getVersionFromURL(database, amazonKeyspacesCAURL)
+	}
+
+	return nil, trace.NotImplemented("%v doesn't support fetching CA version", database)
 }
 
 // downloadFromURL downloads root certificate from the provided URL.
-func (d *realDownloader) downloadFromURL(downloadURL string) ([]byte, error) {
-	resp, err := http.Get(downloadURL)
+func (d *realDownloader) downloadFromURL(downloadURL string) ([]byte, []byte, error) {
```

Comment:
It would be great if we can avoid repeatedly getting (or heading) the same URL multiple times when there are many databases of the same type. Maybe we can use a FnCache or a cached HTTP client or something. What do you guys think?